internal/io/finder/refiner_test.go

```diff
@@ -18,45 +18,75 @@ func TestConvertPathsToAbsPaths(t *testing.T) {
 }
 
 func TestMapFilesToDir(t *testing.T) {
-
 	dirs := []string{
-		"test" + string(os.PathSeparator),
-		"test2" + string(os.PathSeparator),
+		filepath.Join("foo/bar/tree/target/classfiles") + string(os.PathSeparator),
+		filepath.Join("foo/bar/tree/asd/hej/target/classfiles") + string(os.PathSeparator),
+		filepath.Join("foo/bar/tree/asd/fast/target/classfiles") + string(os.PathSeparator),
 	}
 	files := []string{
-		filepath.Join("test/asd"),
-		filepath.Join("test2/basd/qwe"),
-		filepath.Join("test2/test/asd"),
-		filepath.Join("test3/tes"),
+		filepath.Join("foo/bar/tree/pom.xml"),
+		filepath.Join("foo/bar/tree/asd/fast/pom.xml"),
 	}
-	mapFiles := MapFilesToDir(dirs, files)
+	mapFiles := MapFilesToDir(files, dirs)
 	fmt.Println(dirs)
 	fmt.Println(files)
+	fmt.Println(mapFiles)
 
-	assert.Len(t, mapFiles["test"+string(os.PathSeparator)], 1)
-	assert.Len(t, mapFiles["test2"+string(os.PathSeparator)], 2)
+	assert.Len(t, mapFiles["foo/bar/tree/pom.xml"], 2)
+	assert.Len(t, mapFiles["foo/bar/tree/asd/fast/pom.xml"], 1)
 }
 
-func TestFindLongestDirMatch(t *testing.T) {
+func TestFindPomFileMatch(t *testing.T) {
 
-	dirs := []string{"test" + string(os.PathSeparator), "test2" + string(os.PathSeparator)}
-	file := filepath.Join("test", "asd")
-	dirMatch, err := findLongestDirMatch(file, dirs)
+	classDir := filepath.Join("foo/bar/tree/asd/hej/target/classfiles") + string(os.PathSeparator)
+	pomFiles := []string{
+		filepath.Join("foo/bar/tree/pom.xml"),
+		filepath.Join("foo/bar/tree/asd/fast/pom.xml"),
+	}
+	pomFileMatch, err := findPomFileMatch(classDir, pomFiles)
+
+	assert.Equal(t, "foo/bar/tree/pom.xml", pomFileMatch)
+	assert.Nil(t, err)
+}
+
+func TestFindPomFileMatchDifficult(t *testing.T) {
+
+	classDir := filepath.Join("foo/bar/tree/asd/hej/target/classfiles") + string(os.PathSeparator)
+	pomFiles := []string{
+		filepath.Join("foo/bar/tree/pom.xml"),
+		filepath.Join("foo/bar/tree/asd/fast/pom.xml"),
+		filepath.Join("foo/bar/tree/asd/hej/pom.xml"),
+	}
+	pomFileMatch, err := findPomFileMatch(classDir, pomFiles)
 
-	assert.Equal(t, "test"+string(os.PathSeparator), dirMatch)
+	assert.Equal(t, "foo/bar/tree/asd/hej/pom.xml", pomFileMatch)
 	assert.Nil(t, err)
 }
```

Comment:
replace paths with filepath.join( " list of path parts" , this way we assert correctly on windows as well